backend/app/core/service/loader.py

```diff
@@ -73,147 +129,104 @@ def write_metadata(
         )
         db.add(db_event)
 
+    # TODO: are source IDs really necessary on metadata? Perhaps we really do
+    #       want to keep metadata limited to values from the same source as the
+    #       document, but we should validate this assumption.
+
     # sectors
-    for meta in document_with_metadata.sectors:
-        insert_stmt = insert(Sector).values(
-            # parent_id=TODO,
-            name=meta.name,
-            description=meta.description,
-            source_id=db_document.source_id,
-        )
-        do_nothing_stmt = insert_stmt.on_conflict_do_nothing()
-        return_value = db.execute(do_nothing_stmt)
-        if return_value and return_value.inserted_primary_key:
-            meta_id = return_value.inserted_primary_key[0]
-        else:
-            # This is guaranteed to exist
-            existing_sector = (
-                db.query(Sector)
-                .filter(Sector.name == meta.name)
-                .filter(Sector.source_id == db_document.source_id)
-            ).first()
-            meta_id = existing_sector.id  # type: ignore
-
-        db_bridge = DocumentSector(
-            sector_id=meta_id,
+    for sector in document_with_metadata.sectors:
+        # A sector should already exist, so fail if we cannot find it
+        existing_sector = (
```

Comment:
Being that we only ever get the id ... can we not:
```
           sector_id = db.query(Sector.id)
            .filter(Sector.name == sector.name)
            .filter(Sector.source_id == db_document.source_id).scalar()
```

Reply:
done